src/raster/rasterband.rs

```diff
@@ -811,15 +820,21 @@ impl ColorInterpretation {
     }
 }
 
+/// Types of color interpretations for a [`ColorTable`].
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub enum PaletteInterpretation {
+    /// Grayscale
     Gray,
+    /// Red, Green, Blue and Alpha
     Rgba,
+    /// Cyan, Magenta, Yellow and Black
     Cmyk,
+    /// Hue, Lightness and Saturation
     Hls,
 }
 
 impl PaletteInterpretation {
+    /// Instantiates Self from the C API int value of [`GDALPaletteInterp`].
```

Comment:
Nit (also not sure about the hyperlink):

```suggestion
    /// Creates a Rust [`PaletteInterpretation`] from  a C API [`GDALPaletteInterp`] value.
```